cps/kni.c

```diff
@@ -839,43 +890,58 @@ rd_modroute(const struct nlmsghdr *req)
 	switch(rm->rtm_family) {
 	case AF_INET:
 		mnl_attr_parse(req, sizeof(*rm), data_ipv4_attr_cb, tb);
-		attr_get(&update, rm->rtm_family, tb);
+		*err = attr_get(&update, rm->rtm_family, tb);
 		break;
 	case AF_INET6:
 		mnl_attr_parse(req, sizeof(*rm), data_ipv6_attr_cb, tb);
-		attr_get(&update, rm->rtm_family, tb);
+		*err = attr_get(&update, rm->rtm_family, tb);
 		break;
 	default:
 		CPS_LOG(NOTICE, "Unrecognized family in netlink event: %u\n",
 			rm->rtm_family);
-		break;
+		*err = -EAFNOSUPPORT;
+		goto out;
 	}
 
 	if (update.valid) {
 		if (update.type == RTM_NEWROUTE)
-			new_route(&update, cps_conf);
+			*err = new_route(&update, cps_conf);
 		else
-			del_route(&update, cps_conf->gk);
-	}
-
+			*err = del_route(&update, cps_conf->gk);
+	} else
+		*err = 0;
```

Comment:
If an update is not valid and we have `*err` already set (see `switch..case` above), why do we zero `*err` here?

Reply:
This was a bug -- thanks for catching!